### PR TITLE
Allow SSO to be turned on and off in the admin UI

### PR DESCRIPTION
**Pull request by tlovett1**
@johnwatkins0 mind reviewing? This just adds an option in the admin for people to disable SSO.

### CODE REVIEW

File: includes/classes/SSO/SSO.php [hunk 1]
```diff
@@ -43,6 +54,119 @@ public function setup() {
...
+	 * @return array
+	 */
+	public function get_setting() {
+		$setting = ( TENUP_EXPERIENCE_IS_NETWORK ) ? get_site_option( 'tenup_allow_sso', 'yes' ) : get_option( 'tenup_allow_sso', 'yes' );
+
+		return $setting;
+	}
+
+	/**
+	 * Register restrict REST API setting.
```
Comment (johnwatkins0) [inline includes/classes/SSO/SSO.php:121]:
```suggestion
	 * Register 10up SSO setting.
```

File: includes/classes/SSO/SSO.php [hunk 2]
```diff
@@ -43,6 +54,119 @@ public function setup() {
...
+	public function validate_sso_setting( $value ) {
+		if ( in_array( $value, array( 'yes', 'no' ), true ) ) {
+			return $value;
+		}
+
+		return 'yes';
+	}
+
+	/**
+	 * Display UI for restrict REST API setting.
```
Comment (johnwatkins0) [inline includes/classes/SSO/SSO.php:149]:
```suggestion
	 * Display UI for 10up SSO setting.
```

**Review by johnwatkins0 — APPROVED** [review]
Left a couple inline suggestions on comments but otherwise looks good @tlovett1. Tested the branch locally in both single and multisite environments and everything was working.

File: includes/classes/SSO/SSO.php [hunk 3]
```diff
@@ -43,6 +54,119 @@ public function setup() {
...
+				</tr>
+			</tbody>
+		</table>
+		<?php
+	}
+
+	/**
+	 * Get setting
+	 *
+	 * @return array
```
Comment (claytoncollie) [inline includes/classes/SSO/SSO.php:112]:
```suggestion
	 * @return string
```

**Pull request merged**